### DISCUSSION TITLE
Access Denied You do not have permission to sign in. On using callbacks

### DISCUSSION THREAD

**Discussion by Poylar** [nextauthjs/next-auth#8500] [Help] [answered]
I'm trying to store user details in my remote DB on login.
The problem is that if I use callbacks, then I get an ACCESS_DENIED error, if I remove callbacks, then I log in successfully.

I think the reason is that this data is undefined

accessTokenundefined
idToken:undefined

`export const authOptions = {
    secret: process.env.NEXTAUTH_SECRET,

    providers: [
        GoogleProvider({
            clientId: process.env.GOOGLE_CLIENT_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET,
        }),
    ],
    callbacks: {
        async signIn({ user, account, profile }) {
            if (account.provider === 'google') {
                const { accessToken, idToken } = account;
                console.log(
                    '\naccessToken' + accessToken + '\nidToken:' + idToken,
                );
            }
            await ky
                .post('https://next.rue-spots.de/api/login', {
                    body: JSON.stringify({
                        username: user.name,
                        email: user.email,
                    }),
                })
                .json()
                .catch((err) => {
                    console.log(err);
                });
        },
    },
};
`

Can you help me please?

**Comment by Poylar** [comment, marked as answer]
For Future Readers:
add 'return true'

**Reply by techmustrainee** [reply]
Thanks a lot, I was debugging from a long time.
